File: src/main/java/atlix/model/beans/sub/ProductSupplier.java
package atlix.model.beans.sub;

import atlix.model.beans.Product;
import atlix.model.beans.Supplier;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "products_suppliers")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductSupplier {
    @EmbeddedId
    private ProductSupplierId id = new ProductSupplierId();

    @ManyToOne(fetch = FetchType.LAZY)
    @MapsId("productId")
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne(fetch = FetchType.LAZY)
    @MapsId("supplierId") // Mapea el ID del proveedor desde la clave compuesta
    @JoinColumn(name = "supplier_id", nullable = false)
    private Supplier supplier;

    @Column(nullable = false)
    private double price;
}

File: src/main/kotlin/atlix/model/content/ProductSupplier.kt
package atlix.model.content

import atlix.model.beans.ProductBean
import atlix.model.beans.SupplierBean
import jakarta.persistence.Column
import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.MapsId
import jakarta.persistence.Table

@Entity
@Table(name = "product_supplier")
class ProductSupplier(

    @EmbeddedId
    var id: ProductSupplierId = ProductSupplierId(),

    @ManyToOne
    @MapsId("productId")
    @JoinColumn(name = "product_id")
    var product: ProductBean? = null,

    @ManyToOne
    @MapsId("supplierId")
    @JoinColumn(name = "supplier_id")
    var supplier: SupplierBean? = null,

    @Column(name = "purchase_price")
    var purchasePrice: Double = 0.0
)
